### PR TITLE
Intake - Manual/Auto fixes

### PR DESCRIPTION
**Pull request by prawny-boy**
**What this PR does**
Makes the intake controls more user friendly.

**What does it fix/add?**
Adds Autonomous-ish Manual to the intake, #113

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -81,7 +81,7 @@ BreakBeforeTernaryOperators: true
 BreakConstructorInitializers: BeforeColon
 BreakInheritanceList: BeforeColon
 BreakStringLiterals: true
-ColumnLimit:     110
+ColumnLimit:     80
```
Comment (goanna247) [inline .clang-format:84]: 110

File: build.gradle [hunk 2]
```diff
@@ -1,7 +1,7 @@
 plugins {
     id "cpp"
     id "google-test-test-suite"
-    id "edu.wpi.first.GradleRIO" version "2024.2.1"
+    id "edu.wpi.first.GradleRIO" version "2024.1.1"
```
Comment (goanna247) [inline build.gradle:4]: change the version

File: src/main/cpp/Intake.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Intake::Intake(IntakeConfig config) : _config(config) {}
+
+IntakeConfig Intake::GetConfig() {
+  return _config;
+}
+
+void Intake::OnUpdate(units::second_t dt) {
+  switch (_state) {
+    case IntakeState::kIdle: {
+      _config.IntakeMotor.motorController->SetVoltage(0_V);
```
Comment (goanna247) [inline src/main/cpp/Intake.cpp:16]: remove

File: src/main/cpp/Intake.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      if (_config.intakeSensor->Get()) {
```
Comment (goanna247) [inline src/main/cpp/Intake.cpp:17]: this needs to be == false

Comment (goanna247) [reply src/main/cpp/Intake.cpp:17]: or put a ! before it

File: src/main/cpp/Intake.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      _stringStateName = "Idle";
+      _setVoltage = 0_V;
+    } break;
+    case IntakeState::kRaw: {
+      // _config.IntakeMotor.motorController->SetVoltage(_rawVoltage);
+      _stringStateName = "Raw";
+      _setVoltage = _rawVoltage;
+    } break;
+    case IntakeState::kEject: {
+      _config.IntakeMotor.motorController->SetVoltage(-5_V);
```
Comment (goanna247) [inline src/main/cpp/Intake.cpp:29]: remove

File: src/main/cpp/Intake.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      if (_config.intakeSensor->Get() == 0 && _config.magSensor->Get() == 0) {
+        setState(IntakeState::kIdle);
+      }
+      _stringStateName = "Eject";
+      _setVoltage = 7_V;
+    } break;
+    case IntakeState::kHold: {
+      // _config.IntakeMotor.motorController->SetVoltage(0_V);
```
Comment (goanna247) [inline src/main/cpp/Intake.cpp:37]: check if you still have an object in here

File: src/main/cpp/Intake.cpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      _stringStateName = "Hold";
+      _setVoltage = 0_V;
+    } break;
+    case IntakeState::kIntake: {
+      // _config.IntakeMotor.motorController->SetVoltage(5_V);
```
Comment (goanna247) [inline src/main/cpp/Intake.cpp:42]: you need checks in here to sense if you have the object

File: src/main/cpp/Intake.cpp [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      _stringStateName = "Intake";
+      _setVoltage = -7_V;
+    } break;
+    case IntakeState::kPass: {
+      // _config.IntakeMotor.motorController->SetVoltage(5_V);
+      // if (_config.shooterSensor->Get()) {
+      //   setState(IntakeState::kIdle);
+      //   _stringStateName = "Pass";
+      // }
+      _setVoltage = -7_V;
```
Comment (goanna247) [inline src/main/cpp/Intake.cpp:52]: set string state, stop when sensor no longer senses an object

File: src/main/cpp/IntakeBehaviour.cpp [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+    } else {
+      _intake->setRaw(0_V);
+    }
+    _intake->setState(IntakeState::kRaw);
+
+  } else {
+    if (_codriver.GetRightTriggerAxis() > 0.1) {
+      if (_intaking) {
+        _intaking = false;
+        _intake->setState(IntakeState::kIdle);
```
Comment (goanna247) [inline src/main/cpp/IntakeBehaviour.cpp:40]: remove

File: src/main/cpp/IntakeBehaviour.cpp [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+      } else {
+        _intaking = true;
```
Comment (goanna247) [inline src/main/cpp/IntakeBehaviour.cpp:42]: _ejecting = false

File: src/main/cpp/IntakeBehaviour.cpp [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+    if (_codriver.GetLeftTriggerAxis() > 0.1) {
+      if (_ejecting) {
+        _ejecting = false;
+      } else {
+        _ejecting = true;
+        _intaking = false;
+      }
+    }
+
+    if (_intaking) {
```
Comment (goanna247) [inline src/main/cpp/IntakeBehaviour.cpp:55]: this should just set it to intaking

File: src/main/cpp/IntakeBehaviour.cpp [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+      if (_intake->getState() == IntakeState::kHold || _intake->getState() == IntakeState::kPass) {
+        if (_intake->GetConfig().intakeSensor->Get() == 0) {
+          _intake->setState(IntakeState::kIdle);
+          _intaking = false;
+        } else {
+          _intake->setState(IntakeState::kPass);
+        }
+      } else {
+        if (_intake->GetConfig().intakeSensor->Get() == 1) {
```
Comment (goanna247) [inline src/main/cpp/IntakeBehaviour.cpp:64]: the else should just set to idle

File: src/main/cpp/IntakeBehaviour.cpp [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+          _intake->setState(IntakeState::kHold);
+          _intaking = false;
+        } else {
+          _intake->setState(IntakeState::kIntake);
+        }
+      }
+    }
+
+    if (_ejecting) {
+      if (_intake->GetConfig().intakeSensor->Get() == 0) {
```
Comment (goanna247) [inline src/main/cpp/IntakeBehaviour.cpp:74]: you dont need to check your sensor here

**Review by spacey-sooty — APPROVED** [review]

**Pull request merged**